services/src/main/java/org/keycloak/cookie/SecureContextResolver.java

```diff
@@ -0,0 +1,32 @@
+package org.keycloak.cookie;
+
+import java.net.URI;
+
+class SecureContextResolver {
+
+    static boolean isSecureContext(URI uri) {
+        if (uri.getScheme().equals("https")) {
+            return true;
+        }
+
+        String host = uri.getHost();
+        if (host == null) {
```

Comment:
I don't think we have test cases for this one, how can one construct a host-less `URI`?

Reply:
`127.0.0.256` results in `uri.getHost() == null`. A bit weird, but that's what's happening, and there is actually a test case that covers it ;)

Reply:
Ah missed that, thanks!